addons/stock/models/stock_rule.py

```diff
@@ -35,36 +35,143 @@ class ProcurementRule(models.Model):
+        help="""Create Procurement: A procurement will be created in the source location and the system will try to find a rule to resolve it. The available stock will be ignored.
+             Take from Stock: The products will be taken from the available stock.""")
     route_sequence = fields.Integer('Route Sequence', related='route_id.sequence', store=True)
     picking_type_id = fields.Many2one(
         'stock.picking.type', 'Operation Type',
-        required=True,
-        help="Operation Type determines the way the picking should be shown in the view, reports, ...")
-    delay = fields.Integer('Number of Days', default=0)
-    partner_address_id = fields.Many2one('res.partner', 'Partner Address')
+        required=True)
+    delay = fields.Integer('Delay', default=0, help="The expected date of the created transfer will be computed based on this delay.")
+    partner_address_id = fields.Many2one('res.partner', 'Partner Address', help="Address where goods should be delivered. Optional.")
     propagate = fields.Boolean(
         'Propagate cancel and split', default=True,
-        help='If checked, when the previous move of the move (which was generated by a next procurement) is cancelled or split, the move generated by this move will too')
-    warehouse_id = fields.Many2one('stock.warehouse', 'Served Warehouse', help='The warehouse this rule is for')
+        help="When ticked, if the move is splitted or cancelled, the next move will be too.")
+    warehouse_id = fields.Many2one('stock.warehouse', 'Warehouse')
     propagate_warehouse_id = fields.Many2one(
         'stock.warehouse', 'Warehouse to Propagate',
         help="The warehouse to propagate on the created move/procurement, which can be different of the warehouse this rule is for (e.g for resupplying rules from another warehouse)")
+    auto = fields.Selection([
+        ('manual', 'Manual Operation'),
+        ('transparent', 'Automatic No Step Added')], string='Automatic Move',
+        default='manual', index=True, required=True,
+        help="The 'Manual Operation' value will create a stock move after the current one. "
+             "With 'Automatic No Step Added', the location is replaced in the original move.")
+    rule_message = fields.Html(compute='_compute_action_message')
+
+    @api.onchange('picking_type_id')
+    def _onchange_picking_type(self):
+        """ Modify locations to the default picking type's locations source and
+        destination.
+        """
+        self.location_src_id = self.picking_type_id.default_location_src_id.id
+        self.location_id = self.picking_type_id.default_location_dest_id.id
 
-    def _run_move(self, product_id, product_qty, product_uom, location_id, name, origin, values):
+    @api.onchange('route_id', 'company_id')
+    def _onchange_route(self):
+        """ If a route is set and it exists a route company, ensure that the
+        rule's company is the same than the route's company.
+        """
```

Comment:
now that we removed the global rules, the first part of the docstring is outdated